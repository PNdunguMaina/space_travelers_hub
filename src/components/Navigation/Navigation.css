.logo {
  width: 8vw;
  height: auto;
  margin: 5px;
}

.divNav {
  margin-bottom: 17rem;
}
