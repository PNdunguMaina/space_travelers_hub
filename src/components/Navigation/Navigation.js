import { NavLink } from 'react-router-dom';
import planetIcon from '../../images/planetIcon.png';

const Navigation = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/profile',
      text: ' My Profile',
    },
  ];

  return (
    <div>
      <nav>
        <a href="/">
          <img src={planetIcon} alt="logo" />
          <span>Space Travelers&apos; Hub</span>
        </a>
        <ul>
          {links.map((link) => (
            <li key={link.id}>
              <NavLink to={link.path}>
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </div>

  );
};

export default Navigation;
