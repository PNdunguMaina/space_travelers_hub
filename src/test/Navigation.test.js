import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Renderer from 'react-test-renderer';
import Navigation from '../components/Navigation/Navigation';

function MockHeader() {
  return (
    <BrowserRouter>
      <Navigation />
    </BrowserRouter>
  );
}

it('Renders navigation correctly', () => {
  const tree = Renderer
    .create(<MockHeader />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
