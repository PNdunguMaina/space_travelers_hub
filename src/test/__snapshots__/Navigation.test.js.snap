// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders navigation correctly 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <img
        alt="logo"
        src="planetIcon.png"
      />
      <span>
        Space Travelers' Hub
      </span>
    </a>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className=""
          href="/profile"
          onClick={[Function]}
        >
           My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
